drivers/i2s/i2s_ll_stm32.c

```diff
@@ -597,12 +612,35 @@ static void dma_tx_callback(const struct device *dma_dev, void *arg,
 	k_mem_slab_free(stream->cfg.mem_slab, stream->mem_block);
 	stream->mem_block = NULL;
 
+	/*mark the tx sample completion */
+	stream->sample_tx_block_dma_sent++;
+
 	/* Stop transmission if there was an error */
 	if (stream->state == I2S_STATE_ERROR) {
 		LOG_ERR("TX error detected");
 		goto tx_disable;
 	}
 
+	/* check if the tx stop command can be handled */
+	/*if True, the tx stop command can be handled  */
+	if ((stream->state == I2S_STATE_STOPPING) && (status == DMA_STATUS_COMPLETE)) {
+		/* check if all tx samples have been completely handled */
+		/*As stated in zephyr i2s specification, in case of DRAIN command */
+		/* Send all data in the transmit queue and stop the transmission.*/
```

Comment:
I think multiline comment should follow linux' guidelines:
https://www.kernel.org/doc/html/latest/process/coding-style.html#commenting
This is also the case for other changes.